File: src/RidderRedderApi/RidderRedderApi.Models/Knight.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace RidderRedderApi.Models {

    [Table("Knight")]
    public class Knight {

        [Key]
        public int KnightId { get; set; }
        public string AuthId { get; set; }
        public int LandmarkId { get; set; }
        [ForeignKey("LandmarkId")]
        [JsonIgnore]
        public Landmark Landmark { get; set; }
        public int Colour { get; set; }
        public int Level { get; set; }
    }
}

File: src/ridder-redder/src/models/Knight.ts
export class Knight{
    knightId: number;
    authId: string;
    colour: string;
    level: number;
    landmarkId: number;

    constructor(knightId: number, authId: string, colour: string, level: number, landmarkId: number){
        this.knightId = knightId;
        this.authId = authId;
        this.colour = colour;
        this.level = level;
        this.landmarkId = landmarkId;
    }
}
